docarray/array/storage/memory/seqlike.py

```diff
@@ -56,5 +58,7 @@ def __add__(self, other: Union['Document', Sequence['Document']]):
         return v
 
     def extend(self, values: Iterable['Document']) -> None:
-        self._data.extend(values)
-        self._rebuild_id2offset()
+        last_idx = len(self._id2offset)
+        for i, d in enumerate(values):
+            self._id_to_index[d.id] = last_idx + i
+            self._data.append(d)
```

Comment:
this looks slower than the old version. If the intention to play safe, then simply delete `extend` method completely, Python will automatically delegate back to using `for-loop append`. That is, the only purpose to implement `extend` is to have faster method, if one can not come up with one, then there is no need to implement that